vtm/src/org/oscim/renderer/bucket/ExtrusionBucket.java

```diff
@@ -184,50 +185,41 @@ public void addMesh(GeometryBuffer element) {
                 int vtx2 = index[k++] * 3;
                 int vtx3 = index[k++] * 3;
 
-                float vx1 = points[vtx1 + 0];
-                float vy1 = points[vtx1 + 1];
-                float vz1 = points[vtx1 + 2];
+                float[] v1 = new float[3];
+                v1[0] = points[vtx1 + 0];
+                v1[1] = points[vtx1 + 1];
+                v1[2] = points[vtx1 + 2];
 
-                float vx2 = points[vtx2 + 0];
-                float vy2 = points[vtx2 + 1];
-                float vz2 = points[vtx2 + 2];
+                float[] v2 = new float[3];
+                v2[0] = points[vtx2 + 0];
+                v2[1] = points[vtx2 + 1];
+                v2[2] = points[vtx2 + 2];
 
-                float vx3 = points[vtx3 + 0];
-                float vy3 = points[vtx3 + 1];
-                float vz3 = points[vtx3 + 2];
+                float[] v3 = new float[3];
+                v3[0] = points[vtx3 + 0];
+                v3[1] = points[vtx3 + 1];
+                v3[2] = points[vtx3 + 2];
 
                 // Calculate normal for color gradient
-                float ax = vx2 - vx1;
-                float ay = vy2 - vy1;
-                float az = vz2 - vz1;
-
-                float bx = vx3 - vx1;
-                float by = vy3 - vy1;
-                float bz = vz3 - vz1;
-
-                // Vector product (c is at right angle to a and b)
-                float cx = ay * bz - az * by;
-                float cy = az * bx - ax * bz;
-                float cz = ax * by - ay * bx;
-
-                double len = Math.sqrt(cx * cx + cy * cy + cz * cz);
+                float[] normTris = GeometryUtils.normalOfPlane(v1, v2, v3);
```

Comment:
Similarly here can pass an already created same `reuse` array each time, in order to be filled with the results?

Reply:
e.g. [MercatorProjection.project(GeoPoint p, Point reuse)](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/core/MercatorProjection.java#L486)